### DISCUSSION TITLE
await request.body() or json hangs

### DISCUSSION THREAD

**Discussion by nirradi** [fastapi/fastapi#8534] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI, Request


from fastapi import APIRouter


router = APIRouter(tags=['tag'])


@router.post("/dummypath")
async def get_body(request: Request):
    return await request.body()


app = FastAPI(
    title='Service',
    version='0.1.0',
)


app.include_router(router)
```

### Description

Simple request causes this to hang: 
```

from requests import Session

while 1:
    session = Session()
    print("requesting")
    response = session.post(
        'http://docker.for.mac.localhost:8000/dummypath',
        json=list(range(0, 10000)),
        headers={}
    )

```

this will succeed 4,5 times, and then stop.

### Operating System

Linux

### Operating System Details

Docker image, python:3.10.4-slim-buster
serving with uvicorn==0.18.2

### FastAPI Version

0.78.0

### Python Version

Python 3.9.13

**Comment by nirradi** [comment]
this seems to be only happening on docker for mac, could be a very specific networking issue.

**Comment by NickBenthem** [comment]
Hitting this issue as well - both in Docker for Mac - as well as during a CI/CD check in github - so not sure if it's just Docker for Mac

**Comment by bersena911** [comment]
Just had this issue just on mac.

**Comment by rajeshpanda** [comment]
Having this issue on windows 10 with py310 - poetry env.

**Comment by nirradi** [comment]
Potentially a real issue.

**Comment by rajeshpanda** [comment]
Issue is with Starlette actually. Looks like they are yet to find a good approach for exposing the body of the request.

**Comment by pythops** [comment]
any known workaround while waiting  for the starlette to be fixed ?

**Comment by mabdelfattah** [comment]
Just pumping it up :) !

**Comment by alan-nousot** [comment]
Currently facing this issue as well. My middleware can parse the request body fine but when it hits the endpoint handler function the request body hangs with trying request.body() and request.json()

**Reply by RORO-ML** [reply]
same here, workaround: https://github.com/tiangolo/fastapi/issues/394#issuecomment-883524819

**Comment by taegyunum** [comment]
I'm also facing same issue on Mac OS with docker container.

I know that there are some issues with starlette, so most of case (https://github.com/tiangolo/fastapi/issues/394#issuecomment-883524819) workaround @RORO-ML mentioned works fine.
But in Mac OS docker container this solution didn't work.

For me, this approach helped.
- https://github.com/tiangolo/fastapi/discussions/8187#discussioncomment-7321455

**Comment by Redninjale** [comment]
The same issue is happening for me with Windows 11 WSL docker container. I have tried all the work arounds and nothing seems to work. Most of the threads are about middleware, but I have no middleware and simply want to `await request.json()`. If anyone finds a solution please let me know!

**Comment by YuriiMotov** [comment, marked as answer]
Fixed in Starlette 0.28
